// api/mockDataTree.js
export const branches =  [
  {id:'/branch/1', name: 'Пинские ЭС', childMethod: 'getVoltLevels', isFolder: true, isRoot:true},
  {id:'/branch/2', name: 'Пинские ТС', childMethod: 'getVoltLevels', isFolder: true, isRoot:true},
  {id:'/branch/3', name: 'Барановичские ЭС', childMethod: 'getVoltLevels', isFolder: true, isRoot:true}
]

export const voltLevels1 =  [
  {id:'/branch/1/voltLevel/2', name: '220 кВ', childMethod: 'getSubstations', isFolder:true},
  {id:'/branch/1/voltLevel/3', name: '110 кВ', childMethod: 'getSubstations', isFolder:true}
]
export const voltLevels2 =  [
  {id:'/branch/2/voltLevel/5', name: '10 кВ', childMethod: 'getSubstations', isFolder:true}
]
export const voltLevels3 =  [
  {id:'/branch/3/voltLevel/1', name: '330 кВ', childMethod: 'getSubstations', isFolder:true},
  {id:'/branch/3/voltLevel/2', name: '220 кВ', childMethod: 'getSubstations', isFolder:true},
  {id:'/branch/3/voltLevel/3', name: '110 кВ', childMethod: 'getSubstations', isFolder:true}
]

export const substations1 =  [
  {id:'/substation/1', name: 'Пинск 220', childMethod: '', isFolder:false}
]

export const substations2 =  [
  {id:'/substation/2', name: 'Дрогичин', childMethod: '', isFolder:false},
  {id:'/substation/3', name: 'Бижеревичи', childMethod: '', isFolder:false}
]
export const substations3 =  [
  {id:'/substation/4', name: 'Барановичи 330', childMethod: '', isFolder:false}
]

export const substations4 =  [
  {id:'/substation/5', name: 'Ивацевичи 220', childMethod: '', isFolder:false},
  {id:'/substation/6', name: 'Барановичи 220', childMethod: '', isFolder:false}
]

export const substations5 =  [
  {id:'/substation/7', name: 'Остров', childMethod: '', isFolder:false},
  {id:'/substation/8', name: 'Береза 110', childMethod: '', isFolder:false}
]
export const substations6 =  [
  {id:'/substation/9', name: 'РУСН-10 Пинской ТЭЦ', childMethod: '', isFolder:false},
  {id:'/substation/10', name: 'ЗРУ-10 Лунинецкой ТЭЦ', childMethod: '', isFolder:false}
]
